addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/RequestorJobUnitTest.java

```diff
@@ -368,6 +368,36 @@ void shouldSendMessageWithSetValues() throws IOException {
         assertThat(argument.getValue().getRequestBody().toString(), is(equalTo("aaa=bbb&ccc=ddd")));
     }
 
+    @Test
+    void shouldSendHeadersWithSetValues() throws IOException {
```

Comment:
This is a good start.

I'd also suggest adding a test that has:
- a `requests:` and `headers:` section but no actual headers
- a header name (key) but no value
- a header name (key) and empty string (`""`)